test/spec/util/util.js

```diff
@@ -225,4 +225,32 @@ describe('iD.util', function() {
             expect(iD.utilUnicodeCharsTruncated('😎😬😆😵😴😄🙂🤔', 255)).to.eql('😎😬😆😵😴😄🙂🤔');
         });
     });
+
+    describe('utilNormalizeDateString', function() {
+        it('pads dates', function() {
+            expect(iD.utilNormalizeDateString('1970-01-01').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('1970-01-1').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('1970-1-01').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('-1-1-01').value).to.eql('-0001-01-01');
+            expect(iD.utilNormalizeDateString('123').value).to.eql('0123');
+            expect(iD.utilNormalizeDateString('-4003').value).to.eql('-4003'); // beyond displayable year range but still valid
+            expect(iD.utilNormalizeDateString('31337').value).to.eql('31337');
+            expect(iD.utilNormalizeDateString('-31337').value).to.eql('-31337');
+        });
+        it('wraps dates', function() {
+            expect(iD.utilNormalizeDateString('1970-01-00').value).to.eql('1969-12-31');
+            expect(iD.utilNormalizeDateString('1970-12-32').value).to.eql('1971-01-01');
+            expect(iD.utilNormalizeDateString('1970-02-29').value).to.eql('1970-03-01');
+            expect(iD.utilNormalizeDateString('1970-00').value).to.eql('1969-12');
+            expect(iD.utilNormalizeDateString('1970-23').value).to.eql('1971-11'); // no EDTF for now
+        });
+        it('rejects malformed dates', function() {
+            expect(iD.utilNormalizeDateString('1970-01--1')).to.eql(null);
+            expect(iD.utilNormalizeDateString('197X')).to.eql(null); // no EDTF for now
```

Comment:
Validating the `start_date:edtf` and `end_date:edtf` keys should wait until we have dedicated fields for these keys. Parsing EDTF is somewhat more complex than parsing ordinary ISO&nbsp;8601, since we can’t lean on the browser to bear the brunt of parsing them.